features/logging/cluster_logging_operator.feature

```diff
@@ -152,3 +152,77 @@ Feature: cluster-logging-operator related test
+    """
+    Given I delete the clusterlogging instance
+    Then the step should succeed
+    """
+    Given I obtain test data file "logging/clusterlogging/cl_fluentd-buffer_Invalid.yaml"
+    When I run the :create client command with:
+      | f | cl_fluentd-buffer_Invalid.yaml |
+    Then the step should fail
+
+  # @author gkarager@redhat.com
+  # @case_id OCP-33793
+  @admin
+  @destructive
+  Scenario: Expose more fluentd knobs to support optimizing fluentd for different environments
+    Given I obtain test data file "logging/clusterlogging/cl_fluentd-buffer.yaml"
+    And I create clusterlogging instance with:
+      | remove_logging_pods | true                   |
+      | crd_yaml            | cl_fluentd-buffer.yaml |
+    Then the step should succeed
+    Given I wait for the "fluentd" daemon_set to appear up to 300 seconds
+    And <%= daemon_set('fluentd').replica_counters[:desired] %> pods become ready with labels:
+      | logging-infra=fluentd |
+    And I wait for the "fluentd" config_map to appear up to 300 seconds
+    When I run the :extract admin command with:
+      | resource  | configmap/fluentd |
+      | confirm   | true              |
+    Then the step should succeed
+    And evaluation of `File.read("fluent.conf")` is stored in the :fluent_conf clipboard
+    And the expression should be true> cb.fluent_conf.size > 0
+
+  # @author gkarager@redhat.com
+  # @case_id OCP-33894
+  @admin
+  @destructive
+  Scenario: Fluentd optimizing variable changes trigger new deployment
+    Given I obtain test data file "logging/clusterlogging/cl_fluentd-buffer_default.yaml"
+    And I create clusterlogging instance with:
+      | remove_logging_pods | true                           |
+      | crd_yaml            | cl_fluentd-buffer_default.yaml |
+    Then the step should succeed
+    Given I wait for the "fluentd" daemon_set to appear up to 300 seconds
+    And <%= daemon_set('fluentd').replica_counters[:desired] %> pods become ready with labels:
+      | logging-infra=fluentd |
+    And I wait for the "fluentd" config_map to appear up to 300 seconds
+    When I run the :extract admin command with:
+      | resource  | configmap/fluentd |
+      | confirm   | true              |
+    Then the step should succeed
+    And evaluation of `File.read("fluent.conf")` is stored in the :fluent_conf clipboard
+    And the expression should be true> cb.fluent_conf.size > 0
+    When I run the :patch client command with:
+      | resource      | clusterlogging                                                              |
+      | resource_name | instance                                                                    |
+      | p             | {"spec": {"forwarder": {"fluentd": {"buffer": {"flushMode":"immediate"}}}}} |
+      | type          | merge                                                                       |
+    Then the step should succeed
+    Given I wait for the "fluentd" daemon_set to appear up to 300 seconds
```

Comment:
The ds/fluentd should always be there, the CLO won't recreate it after patching cl/instance, it will update the ds/fluentd.